ui/js/models/resultset.js

```diff
@@ -239,10 +239,18 @@ treeherder.factory('ThResultSetModel', ['$rootScope', '$http', '$location',
                     if (errorMsg === 'credentials must be given') {
                         errorMsg = 'Missing Taskcluster credentials! Please log out and back in again.';
                     }
-                    throw new Error(errorMsg);
+                    return $q.reject(new Error(errorMsg));
                 }
                 return $http.get(url).then(function (resp) {
                     let graph = resp.data;
+
+                    // Build a mapping of buildbot buildername to taskcluster tasklabel for bbb tasks
+                    let builderToTask = _.omit(_.invert(_.mapValues(graph, 'task.payload.buildername')), [undefined]);
+
+                    buildernames = _.map(buildernames, function(buildername) {
```

Comment:
Ah yeah, good point that it's a bit confusing just for the sake of a smaller diff. 254 and 255 are currently used because there are still jobs that are straight bb jobs, not through bbb (or at least I'm led to believe that). I'll update this to be a bit easier to read.